tools/im2rec.py

```diff
@@ -199,68 +180,58 @@ def main():
+        help='Whether to also pack multi dimensional label in the record file') 
+    return parser.parse_args()
 
-    # add data_path here
-    # data_path = os.path.abspath(args.root)
-    data_path = args.root
+if __name__ == '__main__':
+    args = parse_args()
     if args.list:
         make_list(args)
     else:
-        # f is just file name original, not a path, here changed it
-        files = [f for f in os.listdir(data_path) if os.path.isfile(os.path.join(data_path, f))]
-        for f in files:
-            if f.startswith(args.prefix) is True and f.endswith('.lst') is True:
-                print ('Creating .rec file from', f, 'in', args.saving_folder)
-                # join f with data_path
-                image_list = read_list(os.path.join(data_path, f))
-                # delete write record and moved it to the __main__ process
-                return (args, image_list, f)
-
-
-if __name__ == '__main__':
-    # here is the __main__ process, which do the write operation original in write_record 
-    (args, image_list, fname) = main()
-    source = image_list
-    tic = [time.time()]
-    try:
-        import multiprocessing
-        # add freeze_support
-        multiprocessing.freeze_support()
-        q_in = [multiprocessing.Queue() for i in range(args.num_thread)]
-        q_out = multiprocessing.Queue(1024)
-        for i in range(len(image_list)):
-            q_in[i % len(q_in)].put(image_list[i])
-        read_process = [multiprocessing.Process(target=read_worker, args=(args, q_in[i], q_out)) \
-                        for i in range(args.num_thread)]
-        for p in read_process:
-            p.start()
-        write_process = multiprocessing.Process(target=write_worker, args=(q_out, fname, args.saving_folder))
-        write_process.start()
-        for p in read_process:
-            p.join()
-        q_out.put(('finish', '', []))
-        write_process.join()
-    except EOFError:
-        print('multiprocessing not available, fall back to single threaded encoding')
-        import Queue
-        q_out = Queue.Queue()
-        fname_rec = fname[:fname.rfind('.')]
-        # remove the change directory operation, change the write record operation controled by write function it self 
-        saving_file = os.path.join(args.saving_folder, fname_rec + '.rec')
-        record = mx.recordio.MXRecordIO(saving_file, 'w')
-        cnt = 0
-        pre_time = time.time()
-        for item in image_list:
-            image_encode(args, item, q_out)
-            if q_out.empty():
-                continue
-            _, s, _ = q_out.get()
-            record.write(s)
-            cnt += 1
-            if cnt % 1000 == 0:
-                cur_time = time.time()
-                print ('time:', cur_time - pre_time, ' count:', cnt)
-                pre_time = cur_time
-    # add total print operation
-    print ('total: ', len(source))
+        files = [fname for fname in os.listdir('.') if os.path.isfile(fname)]

```

Comment:
files = [f for f in os.listdir(args.root) if os.path.isfile(os.path.join(args.root, f))]

would be work